tools/server/server-context.cpp

```diff
@@ -264,7 +264,7 @@ struct server_slot {
     }
 
     int get_n_draft_max() const {
-        if (!can_speculate()) {
+        if (!can_speculate() && !task->params.speculative.use_self) {
```

Comment:
Refactor this slightly:

- Move the `llama_context * ctx_dft` from `struct server_slot` into `common_speculative * spec`, so that now `spec` would manage the `ctx_dft` lifetime
- Avoid direct references to `ctx_dft` - instead reference `spec` where needed.

For example, `can_speculate()` becomes:

```c++
    bool can_speculate() const {
        return !!spec;
    }
```

- This condition here remains simply `if (!can_speculate()) {`